pkg/controller/bucketclass/bucketclass_controller.go

```diff
@@ -52,34 +54,58 @@ func Add(mgr manager.Manager) error {
 
 	// Watch for changes on resources to trigger reconcile
 	err = c.Watch(&source.Kind{Type: &nbv1.BucketClass{}}, &handler.EnqueueRequestForObject{},
-		bucketClassPredicate, &logEventsPredicate)
+		ignoreUnmatchedProvisioner(options.Namespace), bucketClassPredicate, &logEventsPredicate)
 	if err != nil {
 		return err
 	}
 
 	backingStoreHandler := handler.EnqueueRequestsFromMapFunc(func(obj client.Object) []reconcile.Request {
-			return bucketclass.MapBackingstoreToBucketclasses(types.NamespacedName{
-				Name:      obj.GetName(),
-				Namespace: obj.GetNamespace(),
-			})
-		},
+		return bucketclass.MapBackingstoreToBucketclasses(types.NamespacedName{
+			Name:      obj.GetName(),
+			Namespace: obj.GetNamespace(),
+		})
+	},
 	)
-	err = c.Watch(&source.Kind{Type: &nbv1.BackingStore{}}, backingStoreHandler, logEventsPredicate)
+	err = c.Watch(&source.Kind{Type: &nbv1.BackingStore{}}, backingStoreHandler,
+		util.IgnoreIfNotInNamespace(options.Namespace), logEventsPredicate)
 	if err != nil {
 		return err
 	}
 
 	namespaceStoreHandler := handler.EnqueueRequestsFromMapFunc(func(obj client.Object) []reconcile.Request {
-			return bucketclass.MapNamespacestoreToBucketclasses(types.NamespacedName{
-				Name:      obj.GetName(),
-				Namespace: obj.GetNamespace(),
-			})
-		},
+		return bucketclass.MapNamespacestoreToBucketclasses(types.NamespacedName{
+			Name:      obj.GetName(),
+			Namespace: obj.GetNamespace(),
+		})
+	},
 	)
-	err = c.Watch(&source.Kind{Type: &nbv1.NamespaceStore{}}, namespaceStoreHandler, logEventsPredicate)
+	err = c.Watch(&source.Kind{Type: &nbv1.NamespaceStore{}}, namespaceStoreHandler,
+		util.IgnoreIfNotInNamespace(options.Namespace), logEventsPredicate)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
+
+func ignoreUnmatchedProvisioner(noobaaOperatorNamespace string) predicate.Predicate {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return isObjectForProvisioner(e.Object, noobaaOperatorNamespace)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return isObjectForProvisioner(e.Object, noobaaOperatorNamespace)
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return isObjectForProvisioner(e.ObjectNew, noobaaOperatorNamespace)
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return isObjectForProvisioner(e.Object, noobaaOperatorNamespace)
+		},
+	}
+}
+
+func isObjectForProvisioner(obj client.Object, noobaaOperatorNamespace string) bool {
+	noobaaOperatorLabel := "noobaa-operator"
+	return obj.GetLabels()[noobaaOperatorLabel] == noobaaOperatorNamespace || obj.GetNamespace() == noobaaOperatorNamespace
```

Comment:
I think that if the BC has any `noobaa-operator` label, you should not check for `obj.GetNamespace()`

Reply:
This is to ensure that new operators can work properly with the older ones.

For example: Say namespace `a` is running noobaa-operator (pre-PR) and namespace `b` is running nooba-operator (post-PR). If I create a BucketClass in namespace `a` with label `noobaa-operator: b` then both `a` and `b` will try to take ownership of the resource (because `a` is unaware of the changes in the PR) which will cause weird issues. We can resolve the ownership but it would make everything a bit more convoluted. 

This is albeit an edge case though and I do not expect regular users of NooBaa to ever stumble upon it.

Reply:
Ah, I get it. You pointed out that we should be doing what I just said while the code has a bug 🤦 . Fixing this.